release/downloadIstioCandidate.sh

```diff
@@ -6,23 +6,32 @@
 # so it should be pure bourne shell, not bash (and not reference other scripts)
 #
 # The script fetches the latest Istio release candidate and untars it.
-# It's a copy of ../downloadIstio.sh which is for stable releases but lets
+# It's derived from ../downloadIstio.sh which is for stable releases but lets
 # users do curl -L https://git.io/getLatestIstio | ISTIO_VERSION=0.3.6 sh -
 # for instance to change the version fetched.
 
 # This is the latest release candidate (matches ../istio.VERSION after basic
 # sanity checks)
-ISTIO_VERSION=${ISTIO_VERSION:-0.2.12}
 
-NAME="istio-$ISTIO_VERSION"
 OS="$(uname)"
 if [ "x${OS}" = "xDarwin" ] ; then
   OSEXT="osx"
 else
   # TODO we should check more/complain if not likely to work, etc...
   OSEXT="linux"
 fi
-URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istio-${ISTIO_VERSION}-${OSEXT}.tar.gz"
+
+if [ "x${ISTIO_VERSION}" = "x" ] ; then
+  TEMP_FILE="$(mktemp /tmp/istio.latest.XXXX)"
+  curl -L -s -o "$TEMP_FILE" https://api.github.com/repos/istio/istio/releases/latest
+  NAME=istio-$(grep tag_name "$TEMP_FILE" | sed "s/ *\"tag_name\": *\"\(.*\)\",*/\1/")
+  URL=$(grep browser_download_url "$TEMP_FILE" | grep $OSEXT | sed "s/ *\"browser_download_url\": *\"\(.*\)\",*/\1/")
```

Comment:
The doc on the API (https://developer.github.com/v3/repos/releases/#get-the-latest-release) says "Draft releases and prereleases are not returned by this endpoint", so that's why I don't bother to check for draft or pre-release.

Reply:
then we just need to get the version and nothing else ?